client/reducers/users.js

```diff
@@ -14,25 +13,38 @@ export default (state = initialState, action) => {
                 users
             });
         }
-        case SET_CURRENT_USER: {
-            const { username, name, email } = action.payload;
-            return {
-                ...state,
-                currentUser: {
-                    ...state.currentUser,
-                    username,
-                    name,
-                    email
-                }
-            };
-        }
         case SET_USER_ATS: {
-            let currentUser = Object.assign({}, state.currentUser, {
-                ats: action.payload
-            });
+            let { user, ats } = action.payload;
+            let users = state.users.slice();
+            let findUser = state.users.find(
+                stateUser => stateUser.username === user.username
+            );
+            if (findUser) {
+                let currentUser = findUser;
+                let configuredUserAts = currentUser.configured_ats;
+                let updatedAts = [];
+
+                for (let at of ats) {
+                    let atFound = configuredUserAts.find(
+                        configuredAt =>
+                            configuredAt.at_name_id === at.at_name_id
+                    );
+                    if (atFound) {
+                        atFound.active = at.active;
+                        updatedAts.push(atFound);
+                    } else {
+                        updatedAts.push(at);
+                    }
+                }
+
+                let currentUserIdx = users.findIndex(
+                    user => user.id === currentUser.id
+                );
+                users[currentUserIdx].configured_ats = updatedAts;
+            }
```

Comment:
This will be much better when we key user by id! Maybe I'll slip that refactor in a follow up to my current huge feature PR XD